xbmc/interfaces/json-rpc/FileItemHandler.cpp

```diff
@@ -95,7 +95,7 @@ bool CFileItemHandler::GetField(const std::string &field, const CVariant &info,
 
     if (field == "art")
     {
-      if (thumbLoader != NULL && item->GetArt().size() <= 0 && !fetchedArt &&
+      if (thumbLoader != nullptr && !item->GetArt().empty() && !fetchedArt &&
```

Comment:
```
- if (thumbloader != nullptr && ...
+ if (thumbloader && ...
```

Reply:
Not sure I understand the logics of the fix. Art shall be loaded (be call to FillLibraryArt) if there is already art and not if there is no art? Original logics was more like if no art,load art, which feels correct.

Reply:
You might be right. `fetchedArt` controls if artwork has been already fetched from the library. At the moment kodi reaches this point, the default icon was already filled by `FillInDefaultIcons` so art is not empty but `fetchedArt` is `false`. It worked before because the default icon was loaded through `SetIconImage` which saved it in `m_strIcon` and not in the art map. Maybe all we need to do is to completely remove the `GetArt()` check since it the if condition will only be valid if artwork was not yet fetched from the library and it is in fact a library item (`m_iDbId` checks)

Reply:
> Maybe all we need to do is to completely remove the GetArt() check since the if condition will only be valid if artwork was not yet fetched from the library

 Sounds reasonable.